code/game/objects/items/weapons/stunbaton.dm

```diff
@@ -94,6 +94,36 @@
 			return
 		..()
 
+/obj/item/weapon/melee/baton/attackby(var/obj/item/I, mob/user as mob)
+	..()
+	if(istype(I, /obj/item/weapon/bikehorn))
+
+		if (src.type == /obj/item/weapon/melee/baton/cattleprod)
+			return
+
+		user.visible_message("<span class='notice'>[user] starts jamming \a [src] into the mouth of \a [I].</span>",\
+		"<span class='info'>You do your best to jam \the [src] into the mouth of \the [I].</span>")
+
+		if(do_after(user, get_turf(src), 5 SECONDS))
```

Comment:
You should just pass `src` rather than the turf.  Passing the turf will enable weird behaviour like continuing to put the horn on after someone's dragged away/picked up the baton.